must-call-checker/src/main/java/org/checkerframework/checker/mustcall/MustCallAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,181 @@
+package org.checkerframework.checker.mustcall;
+
+import static org.checkerframework.common.value.ValueCheckerUtils.getValueOfAnnotationWithStringArgument;
+
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MemberReferenceTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.util.Elements;
+import org.checkerframework.checker.mustcall.qual.MustCall;
+import org.checkerframework.checker.mustcall.qual.MustCallTop;
+import org.checkerframework.checker.objectconstruction.qual.NotOwning;
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedArrayType;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.type.ElementQualifierHierarchy;
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.BugInCF;
+import org.checkerframework.javacutil.TreeUtils;
+
+/**
+ * The annotated type factory for the must call checker. Primarily responsible for the subtyping
+ * rules between @MustCall annotations.
+ */
+public class MustCallAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
+
+  /** The top annotation. */
+  final AnnotationMirror TOP;
+
+  /** The bottom annotation. */
+  final AnnotationMirror BOTTOM;
+
+  /**
+   * Default constructor matching super. Should be called automatically.
+   *
+   * @param checker the checker associated with this type factory
+   */
+  public MustCallAnnotatedTypeFactory(final BaseTypeChecker checker) {
+    super(checker);
+    TOP = AnnotationBuilder.fromClass(elements, MustCallTop.class);
+    BOTTOM = createMustCall();
```

Comment:
According to the [type hierarchy in `MustCall.java`](https://github.com/kelloggm/object-construction-checker/pull/207/files#diff-2718c67b17c6924728dd0c221267cf949ab92ae7cf0f71ea33c3320bb8128109R19),
`@MustCall({"m1"}) :> @MustCall({"m1", "m2"})`.
That means that the variable `BOTTOM` is not the bottom type.  Maybe you should rename it to `DEFAULT`, and have a new variable `BOTTOM` that really is bottom.

Reply:
I think the [linked type hierarchy comment](https://github.com/kelloggm/object-construction-checker/pull/207/files#diff-2718c67b17c6924728dd0c221267cf949ab92ae7cf0f71ea33c3320bb8128109R19) is incorrect, which is causing confusion.  With `@MustCall({"m1"}) :> @MustCall({"m1", "m2"})`, the subtype could _add_ must-call obligations, which then (unsoundly) wouldn't be checked if a value is assigned to a supertype variable.  Instead, we want `@MustCall({"m1"}) <: @MustCall({"m1", "m2"})`, so subtyping requires a _subset_ of must-call obligations.  I think @kelloggm implemented this correctly (so `@MustCall({})` is indeed the bottom type), but the linked comment is wrong.

Reply:
Manu is right - the comment is wrong (somehow I always seem to make typos when I use subtyping symbols...). `@MustCall({})` is the bottom type (and the default) in the hierarchy.